esp/esp/survey/templatetags/survey.py

```diff
@@ -170,13 +172,21 @@ def histogram(answer_list, format='html'):
     
     #   We have the necessary EPS file, now we do any necessary conversions and include
     #   it into the output.
+    png_filename = "%s.png" % file_base
     if format == 'tex':
-        return '\includegraphics[width=%fin]{%s}' % (image_width, file_name)
+        image_path = os.path.join(tempfile.gettempdir(), png_filename)
     elif format == 'html':
-        image_path = '%s%s.png' % (HISTOGRAM_DIR, file_base)
-        if not os.path.exists(image_path):
-            os.system('gs -dBATCH -dNOPAUSE -dTextAlphaBits=4 -dDEVICEWIDTHPOINTS=216 -dDEVICEHEIGHTPOINTS=162 -sDEVICE=png16m -R96 -sOutputFile=%s %s' % (image_path, file_name))
-        return '<img src="%s.png" />' % ('/media/' + HISTOGRAM_PATH + file_base)
+        image_path = os.path.join(HISTOGRAM_DIR, png_filename)
+    if not os.path.exists(image_path):
+        subprocess.call(['gs', '-dBATCH', '-dNOPAUSE', '-dTextAlphaBits=4',
+                         '-dDEVICEWIDTHPOINTS=216', '-dDEVICEHEIGHTPOINTS=162',
+                         '-sDEVICE=png16m', '-R96',
+                         '-sOutputFile=' + image_path, file_name])
+    if format == 'tex':
+        return '\includegraphics[width=%fin]{%s}' % (image_width, png_filename)
+    if format == 'html':
+        return '<img src="%s" />' % os.path.join('/media/', HISTOGRAM_PATH,
```

Comment:
``os.path.join`` is incorrect here, since it is a url, not a file system path. I'm not sure if there's an equivalent function to use here, but it should be fine to use the original expression.